.github/workflows/python-workflow.yaml

```diff
@@ -0,0 +1,66 @@
+# This workflow is based on the Github documentation workflow template.
+# See: https://github.com/actions/starter-workflows/blob/main/ci/python-package.yml
+
+name: python-build
+
+on:
+  push:
+    branches: 
+      - main
+  pull_request:
+    branches: 
+      - main
+  schedule:
+    # This job runs every Monday at 9am PST.
+    - cron: "0 16 * * 1"
+
+jobs:
+  test-with-dremio:
+    runs-on: ubuntu-latest
+    defaults:
+        run:
+          working-directory: python
+    strategy:
+      max-parallel: 4
+      matrix:
+        python-version: [3.5, 3.6, 3.7, 3.8]
+
+    services:
+      dremio:
+        image: dremio/dremio-oss:latest
+        ports:
+          - 9047:9047
+          - 31010:31010
+          - 32010:32010
+          - 45678:45678
+        env:
+          DREMIO_JAVA_EXTRA_OPTS: "-Ddebug.addDefaultUser=true -Ddremio.eula.disabled=true"
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install flake8 pytest
+          pip install -r requirements.txt
+      - name: Lint with flake8
+        run: |
+          # Stop the build if there are Python syntax errors or undefined names
+          flake8 example.py --count --select=E9,F63,F7,F82 --show-source --statistics
+          # Exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
+          flake8 example.py --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
+      # Below step can be improved to run a Python or Bash script that checks for Dremio's liveness
+      - name: Sleep before test to ensure that Dremio is started
+        run: |
+          sleep 180
```

Comment:
Please ensure there's a ticket to track improving this.

Reply:
Will keep track of it in Jira.